File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/CreateTriggerWorkflowExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class CreateTriggerWorkflowExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("Authorization", "YOUR_API_KEY");

        var triggerPost = new TriggerPost(
            integrationKey: "generic-trigger",
            comment: "example comment",
            instructions: JsonSerializer.Deserialize<List<Dictionary<string, object>>>("""
                [
                    {
                        "kind": "turnFlagOn"
                    }
                ]
            """)
        );

        try
        {
            var response = new FlagTriggersApi(config).CreateTriggerWorkflow(
                projectKey: "projectKey_string",
                environmentKey: "environmentKey_string",
                featureFlagKey: "featureFlagKey_string",
                triggerPost: triggerPost
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling FlagTriggersApi#CreateTriggerWorkflow: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/CreateTriggerWorkflowExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const triggerPost: api.TriggerPost = {
  integrationKey: "generic-trigger",
  comment: "example comment",
  instructions: [
    {
      "kind": "turnFlagOn"
    }
  ],
};

new api.FlagTriggersApi(configuration).createTriggerWorkflow(
  "projectKey_string", // projectKey
  "environmentKey_string", // environmentKey
  "featureFlagKey_string", // featureFlagKey
  triggerPost,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling FlagTriggersApi#createTriggerWorkflow:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
